triggers/simulations/overview.rst

```diff
@@ -0,0 +1,461 @@
+
+For example, to model an external component that randomly creates instances of a ``Cat`` contract once every second, we could write:
+
+.. code-block:: scala
+  def createRandomCat(
+    ledger: ActorRef[LedgerProcess.Message],
+    actAs: Party
+  )(implicit materializer: Materializer): Behavior[Unit] =
+    Behaviors.withTimers[Unit] { timer =>
+      timer.startTimerWithFixedDelay((), 1.second, 1.second)
+
+      Behaviors.receiveMessage { _ =>
+        val randomCat =
+          CreatedEvent(
+            templateId = Some("Cats:Cat"),
+            createArguments = Some(SValue.SRecord(
+              "Cats:Cat",
+              fields = ImmArray("owner", "name"),
+              values = ArrayList(
+                SValue.SParty(actAs),
+                SValue.SInt64(Random.nextLong()),
+              ),
+            )),
+        )
+        val createEvent = LedgerProcess.ExternalAction(CreateContract(randomCat, actAs))
+
+        ledger ! createEvent
+        Behaviors.same
+      }
+    }
+
+where a ``Cat`` template might be defined as:
+
+.. code-block:: none
+  template Cat
+    with
+      owner : Party
+      name : Int
+    where
+      signatory owner
+
+.. note::
+  Currently, it is not possible to model external components that exercise choices on a contract. This is a known limitation.
+
+  Currently, a level of understanding of the low level Speedy machine ``SValue`` is required when defining create or archive events. This is a known limitation.
+
+Trigger Process Component
+-------------------------
+
+A primitive trigger may be thought of as having state machine like behaviour defined by the following Daml code:
+
+- its user defined Daml state - ``state`` say
+- its ``updateState`` Daml function
+
+  - crudely, we may think of this as defining a side effecting function with type ``TriggerMsg => state => state`` (side effects here being queries against the internal trigger ACS)
+- and its ``rule`` Daml function
+
+  - crudely, we may think of this as defining a side effecting function with type ``TriggerMsg => Party => state => Unit`` (side effects here being ledger command submissions).
+
+More complex trigger behaviours may then be thought of as additional layers of code that encapsulate this primitive behaviour - e.g. user state initialization, scheduled messaging such as heartbeats, filtering of received transaction events, etc.
+
+This layered or compositional approach is the basis for understanding how complex trigger processes may be defined from simpler pieces of code.
+
+As many trigger instances can be defined from a single piece of trigger Daml code, primitive trigger processes are implemented using a `factory pattern <https://en.wikipedia.org/wiki/Factory_method_pattern>`_. Typically an instance of a trigger factory is first declared and then trigger instances (as Akka typed actors with type ``Behavior[TriggerProcess.Message]``) may then be created from that factory.
```

Comment:
`may` -> `can`
I tend to default to `can` instead of `may` here, but it's not really all that important